src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -62,40 +62,61 @@ function WorkspaceCategoriesPage({policyCategories, route}: WorkspaceCategoriesP
-                        </Text>
-                        <View style={[styles.p1, styles.pl2]}>
-                            <Icon
-                                src={Expensicons.ArrowRight}
-                                fill={theme.icon}
-                            />
+            Object.values(policyCategories ?? {}).map((value) => {
+                const isDisabled = value.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE;
+                return {
+                    text: value.name,
+                    keyForList: value.name,
+                    isSelected: !!selectedCategories[value.name],
+                    isDisabled,
+                    pendingAction: value.pendingAction,
+                    errors: value.errors ?? undefined,
+                    rightElement: (
+                        <View style={[styles.flexRow, isDisabled && styles.buttonOpacityDisabled]}>
+                            <Text style={[styles.textSupporting, styles.alignSelfCenter, styles.pl2, styles.label]}>
+                                {value.enabled ? translate('workspace.common.enabled') : translate('workspace.common.disabled')}
+                            </Text>
+                            <View style={[styles.p1, styles.pl2]}>
+                                <Icon
+                                    src={Expensicons.ArrowRight}
+                                    fill={theme.icon}
+                                />
+                            </View>
                         </View>
-                    </View>
-                ),
-            })),
-        [policyCategories, selectedCategories, styles.alignSelfCenter, styles.flexRow, styles.label, styles.p1, styles.pl2, styles.textSupporting, theme.icon, translate],
+                    ),
+                };
+            }),
+        [
+            policyCategories,
+            selectedCategories,
+            styles.alignSelfCenter,
+            styles.buttonOpacityDisabled,
+            styles.flexRow,
+            styles.label,
+            styles.p1,
+            styles.pl2,
+            styles.textSupporting,
+            theme.icon,
+            translate,
+        ],
     );
 
-    const toggleCategory = (category: PolicyForList) => {
+    const toggleCategory = (category: PolicyOption) => {
         setSelectedCategories((prev) => ({
             ...prev,
-            [category.value]: !prev[category.value],
+            [category.keyForList]: !prev[category.keyForList],
         }));
     };
 
     const toggleAllCategories = () => {
-        const isAllSelected = categoryList.every((category) => !!selectedCategories[category.value]);
-        setSelectedCategories(isAllSelected ? {} : Object.fromEntries(categoryList.map((item) => [item.value, true])));
+        const isAllSelected = categoryList.every((category) => !!selectedCategories[category.keyForList]);
```

Comment:
Maybe we could just compare the list size here, i.e. if `categoryList.length() === selectedCategories`. Not sure how much that would help with performance.

Reply:
Sure! refactored toggle category function to filter out categories with false values, to keep state more clean, please review

```
    const toggleCategory = (category: PolicyOption) => {
        setSelectedCategories((prev) => {
            if (prev[category.keyForList]) {
                const {[category.keyForList]: omittedCategory, ...newCategories} = prev;
                return newCategories;
            }
            return {...prev, [category.keyForList]: true};
        });
    };
    
   ```